src/pdiagmat.jl

```diff
@@ -10,9 +10,9 @@ struct PDiagMat{T<:Real,V<:AbstractVector} <: AbstractPDMat{T}
         new{T,S}(d,v,inv_v)
 end
 
-function PDiagMat(v::AbstractVector,inv_v::AbstractVector)
+function PDiagMat(v::A,inv_v::B) where {A <: AbstractVector, B <: AbstractVector}
     @check_argdims length(v) == length(inv_v)
-    PDiagMat{eltype(v),typeof(v)}(length(v), v, inv_v)
+    PDiagMat{eltype(v),Union{A,B}}(length(v), v, inv_v)
```

Comment:
I think it's preferable to convert `v` here such that we avoid the a non-concrete type parameter.

Reply:
What would you think of adding an extra type parameter for the inverse? (Like in the code I just pushed)

I'm hesitant to modify type signatures but that would avoid converting ranges to vectors

Reply:
Changing the type parameters makes this PR breaking. I still think we could just remove `inv_diag`. This would also allow us to keep the type signature, and hence the PR would be non-breaking (IMO fields are not part of the public API).

Reply:
Alright, I've pushed commits that naively remove `inv_diag` at the cost of recomputing it whenever it's currently used, or switching to use `diag` instead

To make sure `PDiagMat` is backwards compatible I've also added a custom accessor and constructor, and if these changes make sense I'll update the tests too

Reply:
Oh, I realized that you made this changes only now after I had opened https://github.com/JuliaStats/PDMats.jl/pull/146 (which also changes `ScalMat`). This PR never computes `map(inv, diag)`, so there are no additional allocations (in fact, it reduces allocations since the inverse is not computed anymore when `PDiagMat` is constructed).

Reply:
I think your MR is a better change, so when it merges we can close this one